crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -85,45 +86,61 @@ pub(crate) struct FormatLeadingAlternateBranchComments<'a> {
 
 impl Format<PyFormatContext<'_>> for FormatLeadingAlternateBranchComments<'_> {
     fn fmt(&self, f: &mut PyFormatter) -> FormatResult<()> {
+        let empty_line_condition = if is_blank_line_after_nested_stub_class_enabled(f.context()) {
```

Comment:
Can you explain why this special handling is required here but not for other cases for which `stub_suite_can_omit_empty_line` returns false?

Reply:
This is to handle cases when a class definition is the last statement in a suite. The empty line formatting in suite is a loop which handles only for in between statements i.e., both preceding and following are present. This is required only before an alternate branch because otherwise it'll be handled by the top-level suite formatting.